core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -171,12 +172,12 @@ long routedBasedOnClusterVersion() {
     }
 
     /** returns the primary term active at the time the operation was performed on the primary shard */
-    public long primaryTerm() {
+    long primaryTerm() {
         return primaryTerm;
```

Comment:
can we assert it's non-0? I worried now that people will use this method while we have the true primary term somewhere. Maybe also adapt the java docs and deprecate the method?